elements/lisk-p2p/src/peer_book/peer_book.ts

```diff
@@ -22,46 +22,50 @@ import {
 	DEFAULT_TRIED_BUCKET_SIZE,
 } from '../constants';
 import { ExistingPeerError } from '../errors';
-import { P2PEnhancedPeerInfo, P2PPeerInfo } from '../p2p_types';
+import { P2PEnhancedPeerInfo, P2PPeerInfo, PeerLists } from '../p2p_types';
 import { PEER_TYPE } from '../utils';
 
-import { NewList, NewListConfig } from './new_list';
-import { TriedList, TriedListConfig } from './tried_list';
+import { NewList } from './new_list';
+import { TriedList } from './tried_list';
 
 export interface PeerBookConfig {
-	readonly newListConfig?: NewListConfig;
-	readonly triedListConfig?: TriedListConfig;
+	readonly sanitizedPeerLists: PeerLists;
 	readonly secret: number;
 }
 
 export class PeerBook {
 	private readonly _newPeers: NewList;
 	private readonly _triedPeers: TriedList;
+	private readonly _bannedIps: Set<string>;
+	private readonly _blacklistedIPs: Set<string>;
+	private readonly _seedPeers: ReadonlyArray<P2PPeerInfo>;
+	private readonly _fixedPeers: ReadonlyArray<P2PPeerInfo>;
+	private readonly _whitelistedPeers: ReadonlyArray<P2PPeerInfo>;
+	private readonly _unbanTimers: Array<NodeJS.Timer | undefined>;
+
 	public constructor({
-		newListConfig: newListConfig,
-		triedListConfig: triedListConfig,
+		sanitizedPeerLists: sanitizedPeerLists,
```

Comment:
Why did we remove configs for new and tried peers? with that, we won't able to pass the configuration of new and tried peers from p2p, I mean as of now we don't have in p2p config but this makes strict implementation or else we can also remove the numOfBuckets, bucketSize and peertype from the constructor

Reply:
Those list were optional configuration and we did not use them anywhere, not even for test cases. I also find better able to add/remove Peers into the peerBook only via the add/remove interface where we are expecting them and having conditional checks.